### PR TITLE
Add just-once package

### PR DESCRIPTION
**Pull request by DylanPiercey**
https://github.com/angus-c/just/issues/47

### CODE REVIEW

**Comment by DylanPiercey** [comment]
If any changes are required here, or you have any feedback please let me know :).

File: packages/function-once/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+let i = 0;
+const getFirst = once(() => ++i);
+getFirst(); // 1
+getFirst(); // 1
+*/
+
+function once(fn) {
+  var called, value;
+  return function wrap() {
+    if (called) return value;
```
Comment (angus-c) [inline packages/function-once/index.js]: add curly braces please (sorry, should have a lint rule for this)

File: packages/function-once/package.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "just-once",
+  "version": "1.0.0",
+  "description": "ensure a function can only be invoked once",
```
Comment (angus-c) [inline packages/function-once/package.json]: "create a function that can only be invoked once"

**Comment by DylanPiercey** [comment]
@angus-c your requested changes should be implemented. Let me know if anything further is needed.

File: packages/function-once/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+module.exports = once;
+
+/*
+let i = 0;
+const getFirst = once(() => ++i);
+getFirst(); // 1
+getFirst(); // 1
+*/
+
+function once(fn) {
```
Comment (angus-c) [inline packages/function-once/index.js]: One more thing: can you test that `fn` is a function and throw `new Error("expected a function but got " + fn)`. Then add a couple more tests where you pass undefined or an object and make sure it throws.

e.g. something like
` t.throws(function () {once(undefined)});`

Thanks! and sorry for the delay getting to this.

**Comment by DylanPiercey** [comment]
No problem with the delay, thanks for getting back to me. Just finished those other changes.

File: packages/function-once/index.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  var called, value;
+
+  if (typeof fn !== 'function') {
+    new Error('expected a function but got ' + fn);
```
Comment (angus-c) [inline packages/function-once/index.js]: don't forget the throw :)

`throw new Error('expected a function but got ' + fn);`

Comment (DylanPiercey) [inline packages/function-once/index.js]: Oh, I just assumed that the tests folder was required by reading the file system.
Will update accordingly. 😛

Comment (angus-c) [inline packages/function-once/index.js]: ha yeah

**Comment by angus-c** [comment]
thanks for this!

**Comment by DylanPiercey** [comment]
No problem, I assume your taking over for here? I just fixed it 😛.

**Comment by angus-c** [comment]
Whoops yeah I just realized I jumped the gun. Will fix.

**Comment by DylanPiercey** [comment]
Alright, you just have to add it to the test list (which I missed) and fix the throw statement. Works great after that!

**Comment by angus-c** [comment]
Ha I missed the test list too! All sorted now. Here's the runkit http://anguscroll.com/just/just-once.  
Thanks again!

**Comment by DylanPiercey** [comment]
Wicked, thanks for your help with the PR.

**Pull request merged**